### PR TITLE
Mayk/filter2

### PR DESCRIPTION
**Pull request by MayHyeyeonKim**
## Description
I implemented the filters dialog. 
For **Location**, users can enter a **city, state, or zip code (case-insensitive)**. When the Enter key is pressed, a **chip** is created below. 
For **Status**, users can select **'active,' 'past,' or 'all.'** After clicking the **'Apply filters'** button, data is fetched from the `hospitalInfoService`, displaying the relevant filtered data.

### ui
Screenshot 2024-09-17 at 4 22 33 PM

Screenshot 2024-09-17 at 4 07 07 PM

### Filtering in the hospitalInfoService
Screenshot 2024-09-17 at 4 19 49 PM

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related Tickets & Documents
[1.1.3.1 Display filter criteria on the map UI](https://projects.digitalaidseattle.org/project-tasks-details?recordId=rec8bckzXGKMi88Tp)
[1.1.3.4 Upgrade hospital-service to take filter criteria and return filtered result](https://projects.digitalaidseattle.org/project-tasks-details?recordId=recQy58l8ioOY668n)
[1.1.3.5 Define hospital-service API for filter functionality](https://projects.digitalaidseattle.org/project-tasks-details?recordId=recgCDMmh4I9dWX3H)

## QA Instructions, Screenshots, Recordings

**1. Location Filter:**

Enter a city, state, or zip code in the Location field.
Press Enter to create a chip for the entered criteria.
Ensure chips are displayed correctly below the input.

**2. Status Filter:**

Select 'active,' 'past,' or 'all' from the Status dropdown.
Verify the selected status is correctly highlighted.

**3. Apply Filters:**

Click the 'Apply filters' button.
Confirm that data is fetched from hospitalInfoService and displays the filtered results.

**4. UI Testing:**

Check the UI for responsiveness and correct rendering based on the screenshots provided.
Ensure that when there are many chips, a scrollbar appears, indicating proper handling of overflow.

**5. Error Handling:**

Note that error handling has not yet been implemented.

**6. Discussion Points:**

Observe the behavior when many chips are added.
Determine if there is a need to limit the number of chips through further discussion with the team and the client.

### CODE REVIEW

**Comment by jnakaso** [comment]
Here are changes that should be implemented:
- The filter component needs to store state of users selections  (i.e. if the user types in a location, the next time filter dialog is opened, the location should be displayed.)

**Comment by MayHyeyeonKim** [comment]
> Here are changes that should be implemented:
> 
> * The filter component needs to store state of users selections  (i.e. if the user types in a location, the next time filter dialog is opened, the location should be displayed.)
> * context for Hospitals  (see useContext)  This will simplify the codebase quite a bit by not having to pass hospitals as a argument to every component.  I can show how this can be done.

Thank you for showing me how to use useContext, I just modified that the part and commited it again.
I also saw that you implemented the search functionality in `hospitalInfoService`. It's working well too! 😊👍

**Comment by MayHyeyeonKim** [comment]
added `Clear all` button
Screenshot 2024-10-03 at 1 47 44 AM

**Review by jnakaso — CHANGES_REQUESTED** [review]
Please address various comments

File: src/components/GFLMap.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const GFLMap: React.FC<MapProps> = ({
       <ScaleControl />
       {hospitals.map((hospital) => (
         <Marker
-          color={hospital.status === "Closed" ? "#DB5757" : "#92C65E"}
+          color={hospital.status === "past" ? "#DB5757" : "#92C65E"}
```
[the remark this answers is not in the document]
Comment (MayHyeyeonKim) [reply src/components/GFLMap.tsx]: [question]
I set the status in `hospitalInfo` as a union type (`"active"` | `"past"`). However, in `thumbnailData`, I set the status of an entry to `"blabla"`. Why doesn't it throw an error even though the status in `hospitalInfo` is not one of `"active"` or `"past"`?

[Does this commit align with the method you suggested?](https://github.com/digitalaidseattle/games-for-love/pull/19/commits/d73481c88d61ad7cce456414139f59277f9a3d88)

File: src/components/Filter.tsx
[the remark this answers is not in the document]
Comment (MayHyeyeonKim) [reply src/components/Filter.tsx:1]: https://github.com/digitalaidseattle/games-for-love/pull/19/commits/da8999e2584c661a2204552840eb8ac7f2eff7ea

File: src/components/Filter.tsx [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+  "& .MuiDialog-paper": {
+    width: "400px",
+    height: "480px",
+    maxWidth: "none",
+    margin: "auto",
+    borderRadius: "15px",
+  },
+}));
+
+interface FilterProps {
```
[the remark this answers is not in the document]
Comment (MayHyeyeonKim) [reply src/components/Filter.tsx]: https://github.com/digitalaidseattle/games-for-love/pull/19/commits/da8999e2584c661a2204552840eb8ac7f2eff7ea

File: src/components/Filter.tsx [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+  Button,
+} from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import { styled } from "@mui/material/styles";
+import LocationOnIcon from "@mui/icons-material/LocationOn";
+import { InputAdornment } from "@mui/material";
+import { hospitalInfoService } from "../services/hospitalInfo/hospitalInfoService";
+import { HospitalsContext } from "../context/HospitalsContext";
+
+const BootstrapDialog = styled(Dialog)(() => ({
```
[the remark this answers is not in the document]
Comment (MayHyeyeonKim) [reply src/components/Filter.tsx]: https://github.com/digitalaidseattle/games-for-love/pull/19/commits/da8999e2584c661a2204552840eb8ac7f2eff7ea

File: src/context/HospitalsContext.tsx
[the remark this answers is not in the document]
Comment (MayHyeyeonKim) [reply src/context/HospitalsContext.tsx:1]: https://github.com/digitalaidseattle/games-for-love/pull/19/commits/a1dfa4c6bd130883b167982e7ade5b4b9b8df538

File: src/services/hospitalInfo/hospitalInfoService.ts [hunk 4]
```diff
@@ -12,9 +13,17 @@ class HospitalInfoService {
...
   };
+
+  filterHospitals = (hospitals: HospitalInfo[], searchTerm: string) => {
+    return hospitals.filter(
+      (h: HospitalInfo) =>
+        h.name.toLowerCase().indexOf(searchTerm.toLowerCase()) > -1
+    );
+  };
+
   async getHospitalInfo(): Promise<HospitalInfo[]> {
```
[the remark this answers is not in the document]
Comment (MayHyeyeonKim) [reply src/services/hospitalInfo/hospitalInfoService.ts]: [Is it right?](https://github.com/digitalaidseattle/games-for-love/pull/19/commits/bfeef52f55cd70b0087ec80fee308c18db9f29cb)

File: src/components/Filter.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (MayHyeyeonKim) [reply src/components/Filter.tsx]: I couldnt understand. Could you plz explain it again?

[the remark this answers is not in the document]
Comment (jnakaso) [reply src/components/Filter.tsx]: Imagine we needed a SortOptionsDialog or HospitalInfoDialog.  The type for the dialog could look like  :

```
 interface SortOptionsDialogProps {
   open: boolean;
   handleClose: () => void;
 }
 
 const SortOptionsDialog: React.FC<SortOptionsDialogProps>

 interface HospitalInfoDialogProps {
   open: boolean;
   handleClose: () => void;
 }
 
 const HospitalInfoDialog: React.FC<HospitalInfoDialogProps>
```

The code could be refactored to look like

```
 interface DialogProps {
   open: boolean;
   handleClose: () => void;
 }
 
 const SortOptionsDialog: React.FC<DialogProps> 
 
 const HospitalInfoDialog: React.FC<DialogProps>
```

What we are trying to imply with code is that all of our dialogs behave in a similar manor.

[the remark this answers is not in the document]
Comment (MayHyeyeonKim) [reply src/components/Filter.tsx]: Screenshot 2024-10-07 at 1 26 48 PM

https://github.com/digitalaidseattle/games-for-love/pull/19/commits/f6a09b71641a993de07f000014cc5231b69cab74


Thank you so so much!👍👍👍

**Review by jnakaso — APPROVED** [review]
Looks good.  There are some minor changes we should do, but we'll address them later

**Pull request merged**